Source/JavaScriptCore/wasm/WasmCallee.cpp

```diff
@@ -413,22 +415,28 @@ const StackMap& OptimizingJITCallee::stackmap(CallSiteIndex callSiteIndex) const
 }
 #endif
 
-JSEntrypointInterpreterCallee::JSEntrypointInterpreterCallee(Vector<JSEntrypointInterpreterCalleeMetadata>&& metadata, LLIntCallee* callee)
+JSEntrypointInterpreterCallee::JSEntrypointInterpreterCallee(unsigned frameSize, TypeIndex typeIndex, bool usesSIMD)
     : JSEntrypointCallee(Wasm::CompilationMode::JSEntrypointInterpreterMode)
-    , TrailingArray<JSEntrypointInterpreterCallee, JSEntrypointInterpreterCalleeMetadata>(metadata.size(), metadata.begin(), metadata.end())
-    , wasmCallee(reinterpret_cast<intptr_t>(CalleeBits::boxNativeCalleeIfExists(callee)))
+    , frameSize(frameSize)
+    , typeIndex(typeIndex)
+    , wasmFunctionPrologue(
+        Options::useJIT() ? (usesSIMD ? LLInt::wasmFunctionEntryThunkSIMD().code().retagged<WasmEntryPtrTag>()
+            : LLInt::wasmFunctionEntryThunk().code().retagged<WasmEntryPtrTag>()
+            ) : (usesSIMD ? CodePtr<CFunctionPtrTag>(LLInt::getCodeFunctionPtr<CFunctionPtrTag>(wasm_function_prologue_simd_trampoline)).retagged<WasmEntryPtrTag>()
+                : CodePtr<CFunctionPtrTag>(LLInt::getCodeFunctionPtr<CFunctionPtrTag>(wasm_function_prologue_trampoline)).retagged<WasmEntryPtrTag>())
```

Comment:
Can this logic be broken out into a separate function? It's hard to parse nested ternary operators